### PR TITLE
Add lacking info for palace-info

### PR DESCRIPTION
**Pull request by Huy-Ngo**
This PR will add info about available resamplers for palace.

### CODE REVIEW

File: examples/palace-info.py [hunk 1]
```diff
@@ -37,6 +37,19 @@
 
     print(f'\nInfo of device {args.device.name!r}:')
     print('ALC version: {}.{}'.format(*args.device.alc_version))
+
+    # TODO: AL info
+    with Device() as dev, Context(dev) as ctx:
+        print()
+        print('Available resamplers:')
```
Comment (McSinyx) [inline examples/palace-info.py]: I think this can be in one line.

File: examples/palace-info.py [hunk 2]
```diff
@@ -37,6 +37,19 @@
...
+        default_idx = ctx.default_resampler_index
+        default_resampler = ctx.available_resamplers[default_idx]
+        for resampler in ctx.available_resamplers:
+            if resampler == default_resampler:
```
Comment (McSinyx) [inline examples/palace-info.py]: What is this lol.  Could you also please separate the processing part from the printing part, which I believe will enable you to write this more concisely.

File: examples/palace-info.py [hunk 3]
```diff
@@ -37,6 +37,19 @@
 
     print(f'\nInfo of device {args.device.name!r}:')
     print('ALC version: {}.{}'.format(*args.device.alc_version))
+
+    # TODO: AL info
+    with Device() as dev, Context(dev) as ctx:
```
Comment (McSinyx) [inline examples/palace-info.py]: No need for another device here.

**Review by McSinyx — CHANGES_REQUESTED** [review]
Some early reviews.

File: examples/palace-info.py (hunk 3 above)
Comment (Huy-Ngo) [reply examples/palace-info.py]: Oh wait, `args.device` is already a device, didn't notice that.

**Comment by McSinyx** [comment]
Personally I feel that default annotation is too trivial to have its own function.

**Comment by Huy-Ngo** [comment]
Since I can't go further with this, Is it ok if I merge the branch now and start on `loopback`?

File: examples/palace-info.py [hunk 4]
```diff
@@ -37,6 +37,16 @@
 
     print(f'\nInfo of device {args.device.name!r}:')
     print('ALC version: {}.{}'.format(*args.device.alc_version))
+
+    # TODO: AL info
```
Comment (McSinyx) [inline examples/palace-info.py]: Since this seems to be impossible with alure, please remove the note.

File: examples/palace-info.py [hunk 5]
```diff
@@ -37,6 +37,16 @@
...
+    with Context(args.device) as ctx:
+        print('\nAvailable resamplers:')
+        default_idx = ctx.default_resampler_index
+        resamplers = ctx.available_resamplers
+        resamplers[default_idx] += ' (default)'
+        print(*resamplers, sep='\n')
+        print()
```
Comment (McSinyx) [inline examples/palace-info.py]: Would be nice if you can remove the surrounding newlines and indent the resamplers like the device names.

**Review by McSinyx — CHANGES_REQUESTED** [review]
There's nothing stopping you from working on multiple things at the same time, in fact that'd boost your productivity by avoid wasting stall time thinking about a particular problem.  BTW a few more nits:

File: examples/palace-info.py (hunk 5 above)
Comment (McSinyx) [reply examples/palace-info.py]: The newline above too please.  The idea is for `Info of device ...` to look like a continuous block.  Otherwise LGTM.

**Pull request merged**